panels/models/attraction.py

```diff
@@ -0,0 +1,398 @@
+from sqlalchemy.orm import backref
+from sqlalchemy.schema import ForeignKey, UniqueConstraint
+from sqlalchemy.sql import text
+from sqlalchemy.types import Integer
+
+from uber.config import c
+from uber.custom_tags import humanize_timedelta
+from uber.decorators import validation
+from uber.models import MagModel, Session
+from uber.models.types import default_relationship as relationship, Choice, \
+    DefaultColumn as Column, utcnow
+from uber.utils import ceil_datetime, floor_datetime, noon_datetime, \
+    evening_datetime
+
+
+__all__ = [
+    'Attraction', 'AttractionFeature', 'AttractionEvent', 'AttractionSignup']
+
+
+@Session.model_mixin
+class Attendee:
+    attraction_signups = relationship(
+        'AttractionSignup',
+        backref='attendee',
+        order_by='AttractionSignup.signup_time')
+
+    @property
+    def attraction_features(self):
+        return list({e.feature for e in self.attraction_events})
+
+    @property
+    def attractions(self):
+        return list({e.feature.attraction for e in self.attraction_events})
+
+    def is_signed_up_for_attraction(self, attraction):
+        return attraction in self.attractions
+
+    def is_signed_up_for_attraction_feature(self, feature):
+        return feature in self.attraction_features
+
+    def can_admin_attraction(self, attraction):
+        if not self.admin_account:
+            return False
+        return self.admin_account.id == attraction.owner_id \
+            or self.can_admin_dept_for(attraction.department_id)
+
+
+class Attraction(MagModel):
+    NONE = 0
+    PER_FEATURE = 1
+    PER_ATTRACTION = 2
+    RESTRICTION_OPTS = [(
+        NONE,
+        'None – '
+        'Attendees can attend as many events as they wish '
+        '(least restrictive)'
+    ), (
+        PER_FEATURE,
+        'Once Per Feature – '
+        'Attendees can only attend each feature once'
+    ), (
+        PER_ATTRACTION,
+        'Once Per Attraction – '
+        'Attendees can only attend this attraction once '
+        '(most restrictive)'
+    )]
+    RESTRICTIONS = dict(RESTRICTION_OPTS)
+
+    REQUIRED_CHECKIN_OPTS = [
+        (-1, 'Anytime during event'),
+        (0, 'When the event starts'),
+        (300, '5 minutes before'),
+        (600, '10 minutes before'),
+        (900, '15 minutes before'),
+        (1200, '20 minutes before'),
+        (1800, '30 minutes before'),
+        (2700, '45 minutes before'),
+        (3600, '1 hour before')]
+
+    NOTIFICATIONS_OPTS = [
+        ('', 'Never'),
+        (0, 'When the event starts'),
+        (300, '5 minutes before'),
+        (900, '15 minutes before'),
+        (1800, '30 minutes before'),
+        (3600, '1 hour before'),
+        (7200, '2 hours before'),
+        (86400, '1 day before')]
+
+    name = Column(UnicodeText, unique=True)
+    description = Column(UnicodeText)
+    notifications = Column(JSON, default=[], server_default='[]')
+    required_checkin = Column(Integer, default=0)  # In seconds
```

Comment:
Because I wanted this column to be able to store an arbitrary number of seconds. `REQUIRED_CHECKIN_OPTS` probably should have been declared somewhere else, like maybe directly in the HTML template? (I think that's the only place it's used)